.github/workflows/release.yaml

```diff
@@ -53,34 +57,39 @@ jobs:
           echo $result
           exit 1;
         fi
-    - name: Build image
+    # https://github.com/docker/setup-qemu-action
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    # https://github.com/docker/setup-buildx-action
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+    # List available platforms for buildx
+    - name: Available platforms
+      run: echo ${{ steps.buildx.outputs.platforms }}
+    # Build the image using buildx
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Build and push
       id: build
       if: steps.image_existence.outputs.exists == 'false'
       env:
+        BUILDER_INSTANCE: ${{ steps.buildx.outputs.name }}
         IMAGE_TAG: ${{ steps.image_existence.outputs.tag }}
         KUBECTL_VERSION: ${{ steps.kubectl.outputs.version }}
-        KUSTOMIZE_DOWNLOAD_URL: ${{ steps.kustomize.outputs.url }}
+        KUSTOMIZE_VERSION: ${{ steps.kustomize.outputs.version }}
+        REPO_OWNER: ${{github.repository_owner}}
       run: |
-        docker build . -t line/kubectl-kustomize:${IMAGE_TAG} \
+        docker buildx build \
+          --builder ${BUILDER_INSTANCE} \
+          --tag ${REPO_OWNER}/kubectl-kustomize:${IMAGE_TAG} \
+          --tag ${REPO_OWNER}/kubectl-kustomize:latest \
+          --platform linux/amd64,linux/arm64 \
+          --push \
           --build-arg KUBECTL_VERSION=${KUBECTL_VERSION} \
-          --build-arg KUSTOMIZE_DOWNLOAD_URL=${KUSTOMIZE_DOWNLOAD_URL}
-        docker tag line/kubectl-kustomize:${IMAGE_TAG} line/kubectl-kustomize:latest
+          --build-arg KUSTOMIZE_VERSION=${KUSTOMIZE_VERSION} \
+          .
         echo ::set-output name=tag::${IMAGE_TAG}
-    - name: Validate image
```

Comment:
It's much easier to build and push at the same time when supporting multiple platforms. It's not a hard requirement, there are ways to split them, but rather than jump through hoops I moved the test to the last action during image creation.